app/domain/mail_relay/lists.rb

```diff
@@ -179,5 +181,33 @@ def strip_spam_headers(message)
       end
     end
 
+    def append_unsubscribe
+      if defined? mailing_list
+        if mailing_list.subscribable?
+          html?(message) ? append_unsubscribe_link : append_unsubscribe_text
```

Comment:
usually it's a good idea to extract as much as possible into own methods.
maybe the things you're doing in #append_unsubscribe_link, #append_unsubscribe_text would fit easy in this method here. 

add an early return for mailing_list.subscribable?